File: app/src/main/java/com/example/test_store/Logowanie/LoginPresenter.java
package com.example.test_store.Logowanie;

import android.content.Intent;
import android.text.TextUtils;

import android.view.View;
import android.widget.Toast;

import com.example.test_store.BottomNavigation;
import com.example.test_store.Database.Database;
import com.example.test_store.Database.ResultDataListenerAdapter;
import com.example.test_store.Register.Register;

import static com.example.test_store.Constants.EMAIL_REGEX;
import static com.example.test_store.Constants.PASSWORD_REGEX;

public class LoginPresenter extends ResultDataListenerAdapter implements LoginContract.Presenter{
    private Login view;
    private Database database;

    public LoginPresenter(Login view){
        this.view = view;
        database = new Database();
        database.setListener(this);
        checkIfLoggedIn();
    }

    public LoginPresenter(Login view, Database db) {
        this.view = view;
        database = db;
        database.setListener(this);
        checkIfLoggedIn();
    }

    public void checkIfLoggedIn(){
        if(database.isLoggedIn()){
            view.startActivity(new Intent(view.getApplicationContext(), BottomNavigation.class));
            view.finish();
        }
    }

    public boolean validateEmail(String email){
        return email.matches(EMAIL_REGEX);
    }

    public boolean validatePassword(String password){
        if(TextUtils.isEmpty(password)){
            return false;
        }
        //Minimum eight characters, at least one letter and one number:
        if(!password.matches(PASSWORD_REGEX)){
            return false;
        }
        return true;
    }

    public void onLoginClicked() {
        String email = view.mEmail.getText().toString().trim();
        String password = view.mPassword.getText().toString().trim();

       if(!validateEmail(email)){
           view.mEmail.setError("Wrong Email");
           return;
       }
       if(!validatePassword(password)){
           view.mPassword.setError("Wrong password");
           return;
       }

        view.progressBar.setVisibility(View.VISIBLE);
        database.login(email, password);
    }


    @Override
    public void onLoginListener(boolean isSuccess) {
        if(isSuccess){
            Toast.makeText(view, "Login successful", Toast.LENGTH_SHORT).show();
            view.startActivity(new Intent(view.getApplicationContext(), BottomNavigation.class));
        }
        else{
            //Toast.makeText(view., "Error: " + task.getException(), Toast.LENGTH_SHORT).show();
            view.progressBar.setVisibility(View.GONE);
        }
    }

    @Override
    public void onDataResultListener(String result) {
        view.showToast(result);
    }

    public void openRegisterActivity() {
        view.startActivity(new Intent(view.getApplicationContext(), Register.class));
    }
}

File: app/src/main/java/com/example/test_store/Logowanie/LoginPresenter.kt
package com.example.test_store.Logowanie

import android.content.Intent
import android.text.TextUtils
import android.view.View
import android.widget.Toast
import com.example.test_store.BottomNavigation
import com.example.test_store.Constants
import com.example.test_store.Database.Database
import com.example.test_store.Database.ResultDataListenerAdapter
import com.example.test_store.R
import com.example.test_store.Register.Register

class LoginPresenter : ResultDataListenerAdapter, LoginContract.Presenter {
    private var view: Login
    private var database: Database

    constructor(view: Login) {
        this.view = view
        database = Database()
        database.setListener(this)
        checkIfLoggedIn()
    }

    constructor(view: Login, db: Database) {
        this.view = view
        database = db
        database.setListener(this)
        checkIfLoggedIn()
    }

    override fun checkIfLoggedIn() {
        if (database.isLoggedIn) {
            view.startActivity(Intent(view.applicationContext, BottomNavigation::class.java))
            view.finish()
        }
    }

    override fun validateEmail(email: String?): Boolean {
        if (email != null) {
            return email.matches(Constants.EMAIL_REGEX.toRegex())
        }
        return false
    }

    override fun validatePassword(password: String?): Boolean {
        if (TextUtils.isEmpty(password)) {
            return false
        }
        //Minimum eight characters, at least one letter and one number:
        if (password != null) {
            return password.matches(Constants.PASSWORD_REGEX.toRegex())
        }
        return false
    }

    override fun onLoginClicked() {
        val email: String = view.mEmail.text.toString().trim { it <= ' ' }
        val password: String = view.mPassword.text.toString().trim { it <= ' ' }
        if (!validateEmail(email)) {
            view.mEmail.error = view.getString(R.string.incorrect_email)
            return
        }
        if (!validatePassword(password)) {
            view.mPassword.error = view.getString(R.string.incorrect_password)
            return
        }
        view.progressBar.visibility = View.VISIBLE
        database.login(email, password)
    }

    override fun onLoginListener(isSuccess: Boolean) {
        if (isSuccess) {
            Toast.makeText(view, view.getString(R.string.succesful_login), Toast.LENGTH_SHORT).show()
            view.startActivity(Intent(view.applicationContext, BottomNavigation::class.java))
        } else {
            view.progressBar.visibility = View.GONE
        }
    }

    override fun onDataResultListener(result: String) {
        view.showToast(result)
    }

    override fun openRegisterActivity() {
        view.startActivity(Intent(view.applicationContext, Register::class.java))
    }
}


